PERF: Improve pickle support with BZ2 & LZMA (#49068)
pandas-dev/pandas: doc/source/whatsnew/v2.0.0.rst

- Performance improvement in :func:`merge` and :meth:`DataFrame.join` when joining on a sorted :class:`MultiIndex` (:issue:`48504`)
- Performance improvement in :meth:`DataFrame.loc` and :meth:`Series.loc` for tuple-based indexing of a :class:`MultiIndex` (:issue:`48384`)
- Performance improvement for :meth:`MultiIndex.unique` (:issue:`48335`)

> - Performance use out-of-band pickling with BZ2 & LZMA files (:issue:`49068`)

Author: How does this look?

Author: How about this?

```suggestion
- Minimize memory usage when writing BZ2 & LZMA files by using out-of-band pickling (:issue:`49068`)
```

Author: Went ahead and included this. Though tweaks are still welcome :)

- Performance improvement in :func:`merge` and :meth:`DataFrame.join` when joining on a sorted :class:`MultiIndex` (:issue:`48504`)
- Performance improvement in :meth:`DataFrame.loc` and :meth:`Series.loc` for tuple-based indexing of a :class:`MultiIndex` (:issue:`48384`)
- Performance improvement for :meth:`MultiIndex.unique` (:issue:`48335`)

> - Reduce memory usage of DataFrame/Series `to_pickle` when using BZ2 or LZMA (:issue:`49068`)

Reviewer: Sorry for all the nitpicks, can you please use the patterns used elsewhere:
:meth:`DataFrame.to_pickle` and  :meth:`Series.to_pickle`

Author: ```suggestion
- Reduce memory usage of :meth:`DataFrame.to_pickle`/:meth:`Series.to_pickle` when using BZ2 or LZMA (:issue:`49068`)
```

> - Minimize memory usage when writing BZ2 & LZMA files by using out-of-band pickling (:issue:`49068`)

Reviewer: Would be good to reference DataFrame/Series.to_pickle in the whatsnew

Author: ```suggestion
- Reduce memory usage of DataFrame/Series `to_pickle` when using BZ2 or LZMA (:issue:`49068`)
```
